internal/services/elastic/elastic_tag_rule_resource.go

```diff
@@ -0,0 +1,308 @@
+							Optional: true,
+							Elem: &pluginsdk.Resource{
+								Schema: map[string]*pluginsdk.Schema{
+									"name": {
+										Type:     pluginsdk.TypeString,
+										Required: true,
+									},
+									"value": {
+										Type:     pluginsdk.TypeString,
+										Required: true,
+									},
+									"action": {
+										Type:     pluginsdk.TypeString,
+										Required: true,
+										ValidateFunc: validation.StringInSlice([]string{
+											"Include",
+											"Exclude",
+										}, false),
+									},
+								},
+							},
+						},
+					},
+				},
+			},
+
+			"provisioning_state": {
+				Type:     pluginsdk.TypeString,
+				Computed: true,
+			},
+		},
+	}
+}
+
+func resourceElasticTagRuleCreateorUpdate(d *pluginsdk.ResourceData, meta interface{}) error {
+	subscriptionId := meta.(*clients.Client).Account.SubscriptionId
+	client := meta.(*clients.Client).Elastic.TagRuleClient
+	ctx, cancel := timeouts.ForCreate(meta.(*clients.Client).StopContext, d)
+	defer cancel()
+
+	name := d.Get("name").(string)
+	resourceGroup := d.Get("resource_group_name").(string)
+	ruleSetName := d.Get("rule_set_name").(string)
+
+	id := parse.NewElasticTagRuleID(subscriptionId, resourceGroup, name, ruleSetName).ID()
+
+	existing, err := client.Get(ctx, resourceGroup, name, ruleSetName)
+	if err != nil {
+		if !utils.ResponseWasNotFound(existing.Response) {
+			return fmt.Errorf("checking for existing Elastic Monitor %q (Resource Group %q): %+v", name, resourceGroup, err)
+		}
+	}
+
+	body := elastic.MonitoringTagRules{
+		Properties: &elastic.MonitoringTagRulesProperties{
+			LogRules: expandLogRules(d.Get("log_rules").([]interface{})),
+		},
+	}
+	if _, err := client.CreateOrUpdate(ctx, resourceGroup, name, ruleSetName, &body); err != nil {
+		return fmt.Errorf("configuring Tag Rules on Elastic Monitor %q (Resource Group %q): %+v", name, resourceGroup, err)
+	}
+
+	d.SetId(id)
+	return resourceElasticTagRuleRead(d, meta)
+}
+
+func resourceElasticTagRuleRead(d *pluginsdk.ResourceData, meta interface{}) error {
+	client := meta.(*clients.Client).Elastic.TagRuleClient
+	ctx, cancel := timeouts.ForRead(meta.(*clients.Client).StopContext, d)
+	defer cancel()
+
+	id, err := parse.ElasticTagRuleID(d.Id())
+	if err != nil {
+		return err
+	}
+
+	resp, err := client.Get(ctx, id.ResourceGroup, id.MonitorName, id.TagRuleName)
+	if err != nil {
+		if utils.ResponseWasNotFound(resp.Response) {
+			log.Printf("[INFO] Elastic monitor %q does not exist - removing from state", d.Id())
+			d.SetId("")
+			return nil
+		}
+	}
+
+	d.Set("name", id.MonitorName)
+	d.Set("resource_group_name", id.ResourceGroup)
+	d.Set("rule_set_name", id.TagRuleName)
+
+	if props := resp.Properties; props != nil {
+		if err := d.Set("log_rules", flattenLogRules(props.LogRules)); err != nil {
+			return fmt.Errorf("setting `log_rules`: %+v", err)
+		}
+		d.Set("provisioning_state", props.ProvisioningState)
+	}
+	d.Set("type", resp.Type)
+	d.Set("id", resp.ID)
```

Comment:
Please use `id.ID()` to prevent unexpected changes from server side.

Reply:
updated